Add default exception message to TraceEvent & Update README.md

joshd-7 opened the pull request.

Commit:
Add default exception message in TraceEvent

Added default exception message to TraceEvent when format input is empty to use default message.

Commit:
Update README.md

fixes #17

sirkirby commented on src/Raygun.Diagnostics/RaygunTraceListener.cs:
```diff
@@ -132,6 +133,9 @@ public override void TraceEvent(TraceEventCache eventCache, string source, Trace
     public override void TraceEvent(TraceEventCache eventCache, string source, TraceEventType eventType, int id, string format, params object[] args)
     {
+      var stackTrace = new StackTrace(true).GetFrame(1);
+      if (format == string.Empty) format = $"An unexpected error occurred while calling {stackTrace.GetMethod().Name} in {Path.GetFileName(stackTrace.GetFileName())} at line {stackTrace.GetFileLineNumber()}";
```
Just one minor adjustment. Lets allow `null` as well. so `if (!string.IsNullOrEmpty(format))`.  

so this syntax would be valid 
```c#
Trace.TraceError(null, new { id = "1234", name = "Anon"});
```

Commit:
Modify format check and stacktrace variables

Storing GetMethod in a variable to access once.
Removed System.IO incase it's not compatible in all environments.
Added IsNullOrEmpty check on format.

Commit:
Refactor Code, Add Unit Tests

Moved code to two functions for better resuability. Added execution to MessageFromTraceEvent and MessageFromString so they're used through other TraceEvent/TraceData.
Added Unit Tests

sirkirby commented:
merged. Thanks @JoshDahl7
